src/flint/fmpz.jl

```diff
@@ -826,6 +826,30 @@ function ^(x::ZZRingElem, y::Union{Int, UInt, ZZRingElem})
   end
 end
 
+###############################################################################
+#
+#  Generic powering by an ZZRingElem
+#
+###############################################################################
+
+^(a::T, n::IntegerUnion) where {T<:RingElem} = _generic_power(a, n)
+
+function _generic_power(a, n::IntegerUnion)
```

Comment:
This is not really specific to `ZZRingElem`. I wonder if we should have a `src/generic.jl` file or perhaps `src/generic` dir in which we move such "generic" code (I think we have more examples in this vein).

Anyway, also OK by me to move it here for now. Just food for thought (since you've been touching many of these files, you probably have a better feeling than me whether this idea has merit or not)

Reply:
There are some functions where it is not really clear, where they should go. In the end, I wouldn't know where I would search for a generic powering function, so I would need `git grep` anyway...